src/plugins/data/server/search/routes/bsearch.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import { catchError, first, map } from 'rxjs/operators';
+import { CoreStart, KibanaRequest } from 'src/core/server';
+import { BfetchServerSetup } from 'src/plugins/bfetch/server';
+import { IKibanaSearchResponse, ISearchClient, ISearchOptions } from '../../../common/search';
+import { shimHitsTotal } from './shim_hits_total';
+
+type GetScopedProider = (coreStart: CoreStart) => (request: KibanaRequest) => ISearchClient;
+
+export function registerBsearchRoute(
+  bfetch: BfetchServerSetup,
+  coreStartPromise: Promise<[CoreStart, any, any]>,
```

Comment:
let's avoid `any` usage here also? 
I can imagine a possible bug where someone would add `coreStart[1].somePlugin` later and wouldn't notice they rely on `any`